### PR TITLE
BUG: Fix configuration of CAMB interpolator redshift grid

### LINKED ISSUE #438: CAMB fails for redshift = 0.0
**Describe the bug**
`skypy.power_spectrum.camb` fails to calculate the linear matter power spectrum for the argument `redshift = 0.0`

**To Reproduce**

```python
>>> from astropy.cosmology import Planck18
>>> import numpy as np
>>> from skypy.power_spectrum import camb
>>> wavenumber = np.logspace(-2, 1, 4)
>>> A_s, n_s = 2.2e-9, 0.76

>>> camb(wavenumber, 1e-5, Planck18, A_s, n_s)  # redshift = 1e-5
array([1.15293630e+05, 9.38033341e+03, 4.88948085e+01, 8.95805129e-02])

>>> camb(wavenumber, 0.0, Planck18, A_s, n_s)  # redshift = 0.0
 Error in dverk, x =   14165.300429599007      xend=   14165.300429599007     
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "skypy/power_spectrum/_camb.py", line 88, in camb
    pk_interp = get_matter_power_interpolator(pars,
  File "lib/python3.8/site-packages/camb/camb.py", line 351, in get_matter_power_interpolator
    results = get_results(pars)
  File "lib/python3.8/site-packages/camb/camb.py", line 35, in get_results
    res.calc_power_spectra(params)
  File "lib/python3.8/site-packages/camb/results.py", line 329, in calc_power_spectra
    self.calc_transfers(params, only_transfers=False)
  File "lib/python3.8/site-packages/camb/results.py", line 311, in calc_transfers
    config.check_global_error('calc_transfer')
  File "lib/python3.8/site-packages/camb/_config.py", line 51, in check_global_error
    raise CAMBError(reference + '%s' % err)
camb.baseconfig.CAMBError: Error in Fortran called from calc_transfer:
DVERK error
```

**Desktop (please complete the following information):**
 - OS: macOS 10.14.6
 - Python: 3.8.7
 - SkyPy: 0.4

**Suggested Solution**
I suspect the cause of this failure is that the CAMB interpolator tries to create a redshift grid between the default zmin parameter (which is zero) and `zmax=np.max(redshift)` (which is also zero). We should either specify a different configuration for the interpolator in this corner case or let the user specify the configuration via kwargs.

### PR DESCRIPTION
**Pull request by rrjbca**
## Description
Fixes configuration of the CAMB interpolator redshift grid so that the user can calculate the power spectrum for specifically `redshift = 0.0`. Fixes #438. More generally the exact power spectrum will be calculated if the number of redshifts requested is less than or equal to the CAMB default parameter `nz_step=100`, otherwise the exact power spectrum is interpolated from a grid of `nz_step` redshifts between `np.min(redshift)` and `np.max(redshift)`. 

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/main/CONTRIBUTING.rst)
- [x] Write unit tests
- [ ] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

File: skypy/power_spectrum/tests/test_camb.py [hunk 1]
```diff
@@ -36,3 +36,41 @@ def test_camb():
...
+def test_camb_redshift_zero():
+    """
+    Regression test for #438
+    Test that camb runs succesfully with redshift=0.0
+    """
+    from skypy.power_spectrum import camb
+    redshift = 0.0
+    wavenumber = np.logspace(-4.0, np.log10(2.0), 200)
+    pzk = camb(wavenumber, redshift, Planck15, 2.e-9, 0.965)
+    assert allclose(pzk, test_pzk[0], rtol=1.e-4)
```
Comment (Lucia-Fonseca) [inline skypy/power_spectrum/tests/test_camb.py:51]: When running this, I get the following error:
```CAMBError: Error in Fortran called from calc_transfer:
DVERK error```
The rest of tests pass with no error message.

**Review by Lucia-Fonseca — CHANGES_REQUESTED** [review]
* This works for me, I can compute camb power spectrum at redshift 0.
* All tests pass except `test_camb_redshift_zero()` (it might be some setup problem with my machine)

File: skypy/power_spectrum/tests/test_camb.py (hunk 1 above)
Comment (rrjbca) [reply skypy/power_spectrum/tests/test_camb.py:51]: This looks like exactly the error reported in #438 that the pull request is supposed to resolve. What version/branch have you installed/checked out when running these tests?

Comment (Lucia-Fonseca) [reply skypy/power_spectrum/tests/test_camb.py:51]: You are right. I was working on a condo environment with SkyPy v0.3 for some reason.
It works fine now.

**Review by Lucia-Fonseca — APPROVED** [review]
Good to go.

**Pull request merged**